sagemaker-python-sdk/keras_script_mode_pipe_mode_horovod/tensorflow_keras_CIFAR10.ipynb

```diff
@@ -0,0 +1,618 @@
+    "                       source_dir=source_dir,\n",
+    "                       role=role,\n",
+    "                       framework_version='1.12.0',\n",
+    "                       py_version='py3',\n",
+    "                       hyperparameters=hyperparameters,\n",
+    "                       train_instance_count=2, train_instance_type='ml.p3.8xlarge',\n",
+    "                       tags = [{'Key' : 'Project', 'Value' : 'cifar10'},{'Key' : 'TensorBoard', 'Value' : 'dist'}],\n",
+    "                       metric_definitions=keras_metric_definition,\n",
+    "                       distributions=distributions)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, we'll configure ```wait=False``` if you want to see the output logs change to ```wait=True```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "remote_inputs = {'train' : inputs+'/train', 'validation' : inputs+'/validation', 'eval' : inputs+'/eval'}\n",
+    "estimator_dist.fit(remote_inputs, wait=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Local TensorBoard command\n",
+    "Using TenosrBoard we can compare between the different jobs we ran, the following command will print the tensorboard command.\n",
+    "Run it in any environment where you have TensorBoard installed.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python generate_tensorboard_command.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can install [TensorBoard](https://github.com/tensorflow/tensorboard) locally using `pip install tensorboard`\n",
+    "To access an S3 log dir, you need to configure TensorBoard default region, you can do that by configuaring an environment variable AWS_REGION=region_name, e.g. `export AWS_REGION = 'us-east-1'`\n",
+    "\n",
+    "You can access TensorBoard locally at http://localhost:6006\n",
+    "\n",
+    "Based on the TensorBoard metrics, we can see that:\n",
+    "1. All jobs run for 60 epochs (0 - 59)\n",
+    "2. File mode and Pipe mode run for ~7 minutes - Pipemode doesn't effect training performance\n",
+    "3. Distributed mode run for ~2 minutes\n",
+    "4. All of the jobs trained to a similiar validation accuracy\n",
+    "\n",
+    "This example is runing with a small dataset (179M), for larger datasets, by not copying the data, pipemode can reduce the training time significantly"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploy the trained model\n",
+    "\n",
+    "The deploy() method creates an endpoint which serves prediction requests in real-time.\n",
+    "The model saves keras artifacts, Thus, can't use TensorFlow serving.\n",
```

Comment:
"The model saves keras artifacts, so we can't use TensorFlow serving for deployment.\n"